collector/textfile.go

```diff
@@ -208,20 +210,39 @@ func (c *textFileCollector) Update(ch chan<- prometheus.Metric) error {
 		}
 
 		for _, f := range files {
+			metrics_file_path := filepath.Join(path, f.Name())
 			if !strings.HasSuffix(f.Name(), ".prom") {
 				continue
 			}
 
-			mtime, err := c.processFile(path, f.Name(), ch)
+			mtime, families, err := c.processFile(path, f.Name(), ch)
+			if families != nil {
+				for _, mf := range families {
+					metrics_names_to_files[*mf.Name] = append(metrics_names_to_files[*mf.Name], metrics_file_path)
+					parsed_families = append(parsed_families, mf)
+				}
+			}
 			if err != nil {
 				errored = true
 				level.Error(c.logger).Log("msg", "failed to collect textfile data", "file", f.Name(), "err", err)
 				continue
 			}
 
-			mtimes[filepath.Join(path, f.Name())] = *mtime
+			mtimes[metrics_file_path] = *mtime
+		}
+	}
+
+	for _, mf := range parsed_families {
+		if mf.Help == nil {
```

Comment:
What happens if both files do have a `HELP` string for the metric, but it's different? Even if it's supposed to fail (I guess it would create an error somewhere up the call chain?), could we add a test for that so we can see the expected behavior?

Reply:
With different `HELP` strings, it will indeed create an error higher in the call chain, and depending on `ErrorHandling`, metrics will or will not be returned.